File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.js
function solution(numer1, denom1, numer2, denom2) {
    const numer = (numer1 * denom2) + (numer2 * denom1);
    const denom = denom1 * denom2;
    const getGCD = (a, b) => (b === 0? a: getGCD(b, a%b)); //getGCD(b, a%b)는 재귀
    const gcd = getGCD(numer, denom)
    return [numer/gcd, denom/gcd];
}

File: 프로그래머스/lv0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public static int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = new int[2];
        int numer = 0;
        int denom = 0;
        
        numer = (numer1 * denom2) + (numer2 * denom1);
        denom = denom1 * denom2;
        
        int gdc = getGDC(numer, denom);
         
        answer[0] = numer/gdc;
        answer[1] = denom/gdc;
        return answer;
    }
    
        private static int getGDC(int a, int b) {
            if(a%b == 0){
               return b;
            } 
            return getGDC(b, a%b);
        }
        
}